src/Cron_Command.php

```diff
@@ -85,27 +85,110 @@ public function create( $args, $assoc_args ) {
 			}
+			$schedule = '0 ' . trim( $schedule );
 		}
 
 		$this->validate_command( $command );
 		$command = $this->add_sh_c_wrapper( $command );
 
-		Cron::create( [
-			'site_url' => $site,
-			'command'  => $command,
-			'schedule' => $schedule
-		] );
+		Cron::create(
+			[
+				'site_url' => $site,
+				'command'  => $command,
+				'schedule' => $schedule,
+			]
+		);
 
 
 		$this->update_cron_config();
 
 		EE\Utils\delem_log( 'ee cron add end' );
 	}
 
+	/**
+	 * Ensures given command will not create problem with INI syntax.
+	 * Semicolons and Hash(#) in commands do not work for now due to limitation of INI style config ofelia uses.
+	 * See https://github.com/EasyEngine/cron-command/issues/4.
+	 *
+	 * @param string $command Command whose syntax needs to be validated.
+	 *
+	 * @throws \EE\ExitException
+	 */
+	private function validate_command( $command ) {
+
+		if ( strpos( $command, ';' ) !== false ) {
+			EE::error( 'Command chaining using `;` - semi-colon is not supported currently. You can either use `&&` or `||` or creating a second cron job for the chained command.' );
+		}
+		if ( strpos( $command, '#' ) !== false ) {
+			EE::error( 'EasyEngine does not support commands with #' );
+		}
+	}
+
+	/**
+	 * Adds wrapper of `sh -c` to execute composite commands through docker exec properly.
+	 *
+	 * @param string $command Passed command.
+	 *
+	 * @return string Command with properly added wrapper.
+	 */
+	private function add_sh_c_wrapper( $command ) {
+		if ( strpos( $command, 'sh -c' ) !== false ) {
+			return $command;
+		}
+
+		return "sh -c '" . $command . "'";
+	}
+
+	/**
+	 * Generates cron config from DB
+	 */
+	private function update_cron_config() {
+
+		$config = $this->generate_cron_config();
+		file_put_contents( EE_CONF_ROOT . '/cron/config.ini', $config );
+		EE_DOCKER::restart_container( 'ee-cron-scheduler' );
+	}
+
+	/**
+	 * Generates and returns cron config from DB
+	 */
+	private function generate_cron_config() {
+
+		$config_template = file_get_contents( __DIR__ . '/../templates/config.ini.mustache' );
+		$crons           = Cron::all();
+
+		foreach ( $crons as &$cron ) {
+			$job_type       = 'host' === $cron->site_url ? 'job-local' : 'job-exec';
+			$id             = $cron->site_url . '-' . preg_replace( '/[^a-zA-Z0-9\@]/', '-', $cron->command ) . '-' . EE\Utils\random_password( 5 );
+			$id             = preg_replace( '/--+/', '-', $id );
+			$cron->job_type = $job_type;
+			$cron->id       = $id;
+
+			if ( 'host' !== $cron->site_url ) {
+				$cron->container = $this->site_php_container( $cron->site_url );
+			}
+		}
+
+		$me = new Mustache_Engine();
+
+		return $me->render( $config_template, $crons );
+	}
+
+	/**
+	 * Returns php container name of a site.
+	 *
+	 * @param string $site Name of the site whose container name is needed.
+	 *
+	 * @return string Container name.
+	 */
+	private function site_php_container( $site ) {
+		return str_replace( '.', '', $site ) . '_php_1';
```

Comment:
This could be erroneous at some point of time. Better to use `docker-compose exec php` instead of generating the php container name and then running `docker exec` command over it.